### ISSUE TITLE
OmniSharp.Roslyn.CSharp.Services.Navigation.FindUsagesService No document found

### ISSUE THREAD

**Issue by yousihy** [OmniSharp/omnisharp-vscode#4921]
## Environment data
`dotnet --info` output:
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  12.0
 OS Platform: Darwin
 RID:         osx-x64
 Base Path:   /usr/local/share/dotnet/x64/sdk/6.0.100/

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  6.0.100 [/usr/local/share/dotnet/x64/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/x64/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/x64/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download

VS Code version:
Version: 1.62.3 (Universal)
Commit: ccbaa2d27e38e5afa3e5c21c1c7bef4657064247
Date: 2021-11-17T07:59:13.865Z (2 wks ago)
Electron: 13.5.2
Chrome: 91.0.4472.164
Node.js: 14.16.0
V8: 9.1.269.39-electron.0
OS: Darwin arm64 21.1.0

C# Extension version:
v1.23.16

## OmniSharp log
Starting OmniSharp server at 12/3/2021, 12:29:53 AM
    Target: /Users/yhyoussef/Documents/ReactCourse/Reactivities/Reactivities.sln

OmniSharp server started.
    Path: /Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/run
    PID: 3229

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 12.0.1 (x64)
[info]: OmniSharp.Services.DotNetCliService
        Checking the 'DOTNET_ROOT' environment variable to find a .NET SDK
[info]: OmniSharp.Services.DotNetCliService
        Using the 'dotnet' on the PATH.
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 17.0.0 - "/Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild/Current/Bin/MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 17.0.0 - "/Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild/Current/Bin
            CscToolPath = /Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /Users/yhyoussef/.vscode/extensions/ms-dotnettools.csharp-1.23.16/.omnisharp/1.37.16/omnisharp/.msbuild
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/yhyoussef/Documents/ReactCourse/Reactivities'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in '/Users/yhyoussef/Documents/ReactCourse/Reactivities/Reactivities.sln'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/yhyoussef/Documents/ReactCourse/Reactivities'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/yhyoussef/Documents/ReactCourse/Reactivities' on host 3073.
Received response for /v2/getcodeactions but could not find request.
[warn]: OmniSharp.Roslyn.CSharp.Services.Navigation.FindUsagesService
        No document found. File: /Users/yhyoussef/Documents/ReactCourse/Reactivities/API/Startup.cs.

## Steps to reproduce
1. Install dotnet
2. Install VSCode
3. Install C# extension
4. Open ASP Web API Project
5. Try to get intellisense to work or select Microsoft object from the "using" statement or any other object in the C# document
6. Unable to get intellisense and unable to get library suggestion for a reference

## Expected  behavior
I should be able to see details about the object my mouse is on. I should be able to get suggestion from VSCode about the missing reference

## Actual behavior
VSCode is not giving any option for missing references
Code intellisense is not working at all

**Comment by JoeRobich** [comment]
@yousihy Sorry you are running into trouble with the C# extension. It seems that OmniSharp is opening a Solution file which may not have had any projects added to it. You can add projects to your solution with the `dotnet sln add` command.

If this is not the problem please let me know if this ReactCourse is available as a repo that I can troubleshoot against.

**Comment by yousihy** [comment]
Thank you for you reply. There is a solution created with 4 projects attached to it.

There is no repo for this. but you can recreate it using the below in the Terminal:

```
mkdir Reactivities
cd Reactivities/
dotnet new sln
dotnet new webapi -n API
dotnet new classlib -n Application
dotnet new classlib -n Domain
dotnet new classlib -n Persistence
dotnet sln add API/
dotnet sln add Application/
dotnet sln add Domain/
dotnet sln add Persistence/
cd API
dotnet add reference ../Application/
cd ..
cd Application
dotnet add reference ../Persistence/
dotnet add reference ../Domain/
cd ..
cd Persistence
dotnet add reference ../Domain/
```

Maybe I did something wrong while creating the solution and its projects?

**Comment by yousihy** [comment]
@JoeRobich your comment made me look into the solution again. I opened every project setting and then I opened the main `sln` file. I noticed that there are no references to the 4 projects within the mail solution file. It seems that the below commands did not run properly or maybe I simply forgot to run them:

```
dotnet sln add API/
dotnet sln add Application/
dotnet sln add Domain/
dotnet sln add Persistence/
```

I ran the commands and I can now see the projects added correctly to the sln file and OmniSharp is working perfectly again.

Thank you so much for the tip. You saved me hours of troubleshooting this :)

I am closing this issue now as it is not an actual issue
